themes/default/content/blog/typescript-versions/index.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Run Pulumi with Any TypeScript Version"
+
+date: 2024-04-15T10:28:28+02:00
+
+draft: false
+
+meta_desc: The latest version of Pulumi's Node.js SDK now lets you easily pick the version of TypeScript to use.
+
+meta_image: meta.png
+
+authors:
+    - julien-poissonnier
+
+tags:
+    - typescript
+    - nodejs
+    - user-experience
+
+---
+
+Pulumi allows you to use the best features of programming languages to build your infrastructure while ensuring compatibility with your current development environments. To this end Pulumi provides a seamless experience when authoring TypeScript programs, managing the compilation of your program for you. With the latest release of our Node.js SDK, we've made this even more flexible by making it easier than ever to choose the version of TypeScript you want to use.
+
+<!--more-->
+
+Each new release of TypeScript comes with new features and improved type checking, as well as support for the latest JavaScript features. Using the latest version of TypeScript provides you with the maximum flexibility for creating types and gives you more confidence that your program will behave as expected.
+
+Until now, Pulumi's seamless experience was limited to using Pulumi's builtin TypeScript version 3.8. If you wanted to use a more recent version, you had to introduce a build step to your project and let Pulumi run in [plain Javascript mode](https://www.pulumi.com/docs/languages-sdks/javascript/#disabling-built-in-typescript-support). With the release of the [Pulumi Node.js SDK 3.113.0](https://www.npmjs.com/package/@pulumi/pulumi) you can now use any version of TypeScript by adding it as a dependency to your project's `package.json` file.
+
+To upgrade your project to the latest versions of TypeScript and Pulumi, run:
```

Comment:
I believe we are also planning on updating our templates. Let’s do that before the blog post and then reference that you can use the latest version of TypeScript by default in a new Pulumi TypeScript project.